src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,233 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const currentNodeId = useRef();
+  const cursor = useRef(0);
   const selectedItem = useRef();
+  const previousSelectedPath = useRef(selectedPath);
   const theme = useContext(ThemeContext);
+  const visibleNodes = [];
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
+  /**
+   * Assigns focus to current node.
+   */
+  const focusCurrentNode = () => {
+    const currentNode = currentNodeId.current ? document.getElementById(currentNodeId.current) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
 
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   * Ensures that the cursor is synched with the currently selected item.
+   */
   useEffect(() => {
+    const nodeId = selectedItem.current?.getAttribute('id');
+    const idx = visibleNodes.findIndex((el) => el.id === nodeId);
+    if (idx >= 0) {
+      cursor.current = idx;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+    }
     if (selectedItem && selectedItem.current) {
       selectedItem.current.scrollIntoView();
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  useLayoutEffect(() => {
-    if (!isNewSelectedPath) {
-      return;
+  useEffect(() => {
+    const nodeId = selectedItem.current?.getAttribute('id');
+    const idx = visibleNodes.findIndex((el) => el.id === nodeId);
+    if (idx >= 0) {
+      cursor.current = idx;
+      currentNodeId.current = visibleNodes[cursor.current].id;
     }
```

Comment:
This logic appears to be duplicated in the previous effect. Is the same code in the prior useEffect necessary if this effect is also going to be called on initial mount? I'd expect the scroll into view to remain in a separate effect that only happens on mount

Reply:
Yes, in the case of the first effect, we want to update the cursor to match the currently selected item in the case that the user directly navigated to a link via a url. In the case of the second effect, we want to update the cursor in the case that the user selected a new item through mouse clicks.